trieste/acquisition/function.py

```diff
@@ -509,6 +512,156 @@ def prepare_acquisition_function(
         return lambda at: expected_improvement(objective_model, eta)(at) * constraint_fn(at)
 
 
+class ExpectedHypervolumeImprovement(SingleModelAcquisitionBuilder):
+    """
+    Builder for the :func:`expected_hv_improvement` acquisition function
+    The implementation of the acquisition function largely
+    follows :cite:`yang2019efficient`
+    """
+
+    def __repr__(self) -> str:
+        return "ExpectedHypervolumeImprovement()"
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model over the specified ``dataset``.
+        :return: The expected_hv_improvement function.
+        """
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        mean, _ = model.predict(dataset.query_points)
+
+        _pf = Pareto(mean)
+        _reference_pt = get_reference_point(_pf.front)
+        return lambda at: expected_hv_improvement(model, _pf, _reference_pt)(at)
+
+
+def expected_hv_improvement(
+    model: ProbabilisticModel,
+    pareto: Pareto,
+    reference_point: TensorType,
+) -> AcquisitionFunction:
+    r"""
+    HV calculation using Eq. 44 of :cite:`yang2019efficient` paper
+    The expected hypervolume calculation is performed in the non-dominated region, which,
+    provided the partitioned cell, can be reformulated as a combination of several one
+    dimensional generalized expected improvement based on each of the partitioned cell.
```

Comment:
Do you mean "each of the partitioned cells" or "each of the partitioned slices" maybe?

```suggestion
    dimensional generalized expected improvements based on each of the partitioned cell.
```

Reply:
Yes, will modify the doc to make it more clear.